src/libcore/time.rs

```diff
@@ -224,6 +224,49 @@ impl Duration {
         }
     }
 
+    /// Creates a new `Duration` that spans no time.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(duration_zero)]
+    /// use std::time::Duration;
+    ///
+    /// let duration = Duration::zero();
+    /// assert!(duration.is_zero());
+    ///
+    /// const IMMEDIATELY: Duration = Duration::zero();
+    /// assert!(IMMEDIATELY.is_zero());
+    /// ```
+    #[unstable(feature = "duration_zero", issue = "none")]
+    #[inline]
+    pub const fn zero() -> Duration {
+        Duration { secs: 0, nanos: 0 }
+    }
```

Comment:
Please put this whole method directly after `new` (and before `from_secs`). I think that's a better spot when looking at the documentation.